oligo_designer_toolsuite/oligo_specificity_filter/_filter_bowtie.py

```diff
@@ -0,0 +1,209 @@
+    If ``None`` then the whole sequence is considered, defaults to None
+    :type mismatch_region: int
+    """
+
+    def __init__(self, dir_specificity, num_mismatches=3, mismatch_region=None):
+
+        super().__init__(dir_specificity)
+
+        if num_mismatches > 3:
+            raise ValueError(
+                "Choice of min_mismatches out of range for bowtie allignment tool. Please choose a value no greater than 3"
+            )
+        else:
+            self.num_mismatches = num_mismatches
+
+        if mismatch_region < 5:
+            raise ValueError(
+                "Choice of mismatch_region out of range for bowtie allignment tool. Please choose a value no less than 5"
+            )
+        else:
+            self.mismatch_region = mismatch_region
+
+        self.dir_bowtie = os.path.join(self.dir_specificity, "bowtie")
+        Path(self.dir_bowtie).mkdir(parents=True, exist_ok=True)
+
+        self.dir_fasta = os.path.join(self.dir_specificity, "fasta")
+        Path(self.dir_fasta).mkdir(parents=True, exist_ok=True)
+
+    def apply(self, oligo_DB, file_reference_DB, n_jobs):
+        """Apply the bowtie filter in parallel on the given ``oligo_DB``. Each jobs filters a single gene, and  at the same time are generated at most ``n_job`` jobs.
+        The filtered database is returned.
+
+        :param oligo_DB: database containing the probes and their features
+        :type oligo_DB: dict
+        :param file_reference_DB: path to the file that will be used as reference for the alignement
+        :type file_reference_DB: str
+        :param n_jobs: number of simultaneous parallel computations
+        :type n_jobs: int
+        :return: probe info of user-specified genes
+        :rtype : dict
+        """
+        # Some bowtie initializations, change the names
+        index_exists = False
+        index_name = os.path.basename(file_reference_DB)
+        # Check if bowtie index exists
+        for file in os.listdir(self.dir_bowtie):
+            if re.search(f"^{index_name}.*", file):
+                index_exists = True
+                break
+
+        # Create bowtie index if none exists
+        if not index_exists:
+            command1 = "bowtie-build " + file_reference_DB + " " + index_name
+            process = Popen(command1, shell=True, cwd=self.dir_bowtie).wait()
+
+        genes = list(oligo_DB.keys())
+        filtered_oligo_DBs = Parallel(n_jobs=n_jobs)(
+            delayed(self._run_bowtie)(oligo_DB[gene], gene, index_name)
+            for gene in genes
+        )
+
+        # reconstruct the oligos_DB
+        for gene, filtered_oligo_DB in zip(genes, filtered_oligo_DBs):
+            oligo_DB[gene] = filtered_oligo_DB
+
+        return oligo_DB
+
+    def _run_bowtie(self, gene_DB, gene, index_name):
+        """Run Bowtie alignment tool to find regions of local similarity between sequences, where sequences are probes and transcripts.
```

Comment:
Bowtie identifies all **alignments** between...
:param gene: id of **the** gene processed